x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java

```diff
@@ -305,39 +305,58 @@ private ObjectPath getWatchHistoryEntry(String watchId) throws Exception {
+                    }
+                    builder.endArray().endObject().endObject();
+                    builder.startArray("sort").startObject().startObject("trigger_event.triggered_time").field("order", "desc").endObject()
                         .endObject().endArray();
-                builder.endObject();
-
-                Request searchRequest = new Request("POST", "/.watcher-history-*/_search");
-                searchRequest.addParameter(TOTAL_HITS_AS_INT_PARAM, "true");
-                searchRequest.setJsonEntity(Strings.toString(builder));
-                Response response = client().performRequest(searchRequest);
-                ObjectPath objectPath = ObjectPath.createFromResponse(response);
-                int totalHits = objectPath.evaluate("hits.total");
-                assertThat(totalHits, is(greaterThanOrEqualTo(1)));
-                String watchid = objectPath.evaluate("hits.hits.0._source.watch_id");
-                assertThat(watchid, is(watchId));
-                objectPathReference.set(objectPath);
-            } catch (ResponseException e) {
-                final String err = "Failed to perform search of watcher history";
-                logger.info(err, e);
-                fail(err);
+                    builder.endObject();
+
+                    Request searchRequest = new Request("POST", "/.watcher-history-*/_search");
+                    searchRequest.addParameter(TOTAL_HITS_AS_INT_PARAM, "true");
+                    searchRequest.setJsonEntity(Strings.toString(builder));
+                    Response response = client().performRequest(searchRequest);
+                    ObjectPath objectPath = ObjectPath.createFromResponse(response);
+                    int totalHits = objectPath.evaluate("hits.total");
+                    assertThat(totalHits, is(greaterThanOrEqualTo(1)));
+                    String watchid = objectPath.evaluate("hits.hits.0._source.watch_id");
+                    assertThat(watchid, is(watchId));
+                    objectPathReference.set(objectPath);
+                } catch (ResponseException e) {
+                    final String err = "Failed to perform search of watcher history";
+                    logger.info(err, e);
+                    fail(err);
+                }
+            });
+        } catch (AssertionError ae) {
+            {
+                Request request = new Request("GET", "/_watcher/stats");
+                Response response = client().performRequest(request);
+                logger.info("watcher_stats: {}", EntityUtils.toString(response.getEntity()));
             }
-        });
+            {
+                Request request = new Request("GET", "/_cluster/state");
+                Response response = client().performRequest(request);
+                logger.info("cluster_state: {}", EntityUtils.toString(response.getEntity()));
+            }
+            {
+                Request request = new Request("GET", "/.watcher-history-*/_search");
```

Comment:
When this fails and is stuck for during the assertBusy, may result in alot of .watcher_history... should we increase the default size ?